File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> res=new ArrayList<>();
        while(head!=null){
            res.add(head.val);
            head=head.next;
        }
        int l=0;
        int r=res.size()-1;
        while(l<r){
            if(res.get(l)!=res.get(r)){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
    
    
}

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        res=[]
        while (head):
            res.append(head.val)
            head=head.next
        l=0
        r=len(res)-1
        while(l<r):
            if(res[l]!=res[r]):
                return False
            l+=1
            r-=1
        return True
        